File: task2/task2/src/main/java/com/task2/Clothing.java
package com.task2;

public class Clothing extends Product{
    private double width;
    private double length;
    private String sizeClassification;

    public Clothing(){

    }

    public Clothing(int id, String name, double price, String category, int volume, double w, double l, String size){
        super(id, name, price, category, volume);
        width = w;
        length = l;
        sizeClassification = size;
    }

    @Override
    public void display_product_info(){
        super.display_product_info();
        System.out.println("Width: "+getWidth());
        System.out.println("Length: "+getLength());
        System.out.println("Size Classification: "+getSize());
    }

    // Setters
    public void setWidth(double w){
        width = w;
    }

    public void setLength(double l){
        length = l;
    }

    public void setSize(String size){
        sizeClassification = size;
    }

    // Getters
    public double getWidth(){
        return width;
    }

    public double getLength(){
        return length;
    }

    public String getSize(){
        return sizeClassification;
    }
}

File: task3/task3c/Clothing.cpp
#include "Clothing.h"
#include <iostream>

Clothing::Clothing(){}

Clothing::Clothing(int id, const string& name, double price, const string& category, int volume, double w, double l, const string& size): Product(id, name, price, category, volume), width(w), length(l), sizeClassification(size) {}

void Clothing::display_product_info() {
    Product::display_product_info();
    cout << "Width: " << width << endl;
    cout << "Length: " << length << endl;
    cout << "Size Classification: " << sizeClassification << endl;
}

// Setters
void Clothing::setWidth(double w){
    width = w;
}

void Clothing::setLength(double l){
    length = l;
}

void Clothing::setSize(const string& size){
    sizeClassification = size;
}


// Getters
double Clothing::getWidth() const {
    return width;
}

double Clothing::getLength() const {
    return length;
}

string Clothing::getSize() const {
    return sizeClassification;
}